File: backend/api/src/main/kotlin/me/ricoschulz/valueObject/Provider.kt
package me.ricoschulz.valueObject

import kotlin.reflect.KSuspendFunction1
import kotlinx.coroutines.Deferred
import kotlinx.coroutines.DelicateCoroutinesApi
import me.ricoschulz.crawler.asmodee
import me.ricoschulz.crawler.blueBrixx
import me.ricoschulz.crawler.fShop
import me.ricoschulz.crawler.fantasyFlightGames
import me.ricoschulz.crawler.railSim
import me.ricoschulz.crawler.tsw3
import me.ricoschulz.crawler.ulissesSpiele
import me.ricoschulz.crawler.xboxDynasty

@DelicateCoroutinesApi
enum class Provider(
    val crawler: KSuspendFunction1<(News) -> Unit, Deferred<Unit>>,
) {
    ASMODEE(::asmodee), // change for hash
    FANTASY_FLIGHT_GAMES(::fantasyFlightGames),
    ULISSES_SPIELE(::ulissesSpiele),
    F_SHOP(::fShop),
    // DMD_WIZAR("https://dnd.wizards.com/news", dndWizard)
    TSW3(::tsw3),
    BLUE_BRIXX(::blueBrixx),
    XBOX_DYNASTY(::xboxDynasty),
    RAIL_SIM(::railSim)
    // youtube api
}

File: server/src/Domains/Article/ValueObjects/Provider.php
<?php declare(strict_types=1);

namespace Domains\Article\ValueObjects;

enum Provider: string
{
    case RAIL_SIM = 'rail_sim';
    case ASMODEE = 'asmodee';
    case XBOX_DYNASTY = 'xbox_dynasty';
    case TSW = 'tsw';
    case ULISSES_SPIELE = 'ulisses_spiele';
    case F_SHOP = 'f_shop';
    case BLUE_BRIXX = 'blue_brixx';
    case FANTASY_FLIGHT_GAMES = 'fantasy_flight_games';

    public static function getAllValues(): array
    {
        return array_column(Provider::cases(), 'value');
    }
}
